source/initial_temperature/interface.cc

```diff
@@ -146,12 +177,43 @@ namespace aspect
     Manager<dim>::initial_temperature (const Point<dim> &position) const
     {
       double temperature = 0.0;
-      for (typename std::list<std_cxx11::shared_ptr<InitialTemperature::Interface<dim> > >::const_iterator
-           initial_temperature_object = initial_temperature_objects.begin();
+      int i = 0;
+
+      for (typename std::list<std_cxx11::shared_ptr<InitialTemperature::Interface<dim> > >::const_iterator initial_temperature_object = initial_temperature_objects.begin();
            initial_temperature_object != initial_temperature_objects.end();
            ++initial_temperature_object)
         {
-          temperature += (*initial_temperature_object)->initial_temperature(position);
+          switch (model_operators[i])
+            {
+              case add:
+              {
+                temperature += (*initial_temperature_object)->initial_temperature(position);
+                break;
+              }
+              case subtract:
+              {
+                temperature -= (*initial_temperature_object)->initial_temperature(position);
+                break;
+              }
+              case minimum:
+              {
+                temperature = std::min (temperature, (*initial_temperature_object)->initial_temperature(position));
+                break;
+              }
+              case maximum:
+              {
+                temperature = std::max (temperature, (*initial_temperature_object)->initial_temperature(position));
```

Comment:
There is a slight problem here, because `max` and `min` are non-commutative. In the current form the result of some operations depend on the order in the input file. However, I do not see a nice solution for it.